Tests/CoreAPITests/WordPressOrgXMLRPCApiTests.swift

```diff
@@ -59,8 +63,12 @@ class WordPressOrgXMLRPCApiTests: XCTestCase {
             },
             failure: { (error, _) in
                 expect.fulfill()
-
+                // When building for SPM, the compiler doesn't generate the domain constant.
+#if SWIFT_PACKAGE
+                XCTAssertEqual(error.domain, "CoreAPI.WordPressOrgXMLRPCApiError")
+#else
                 XCTAssertEqual(error.domain, WordPressOrgXMLRPCApiErrorDomain)
+#endif
```

Comment:
@crazytonyli what do you think of this?

I worry it might backfire in the real world if people where to do a comparison on the domain with a `String` literal value.

Maybe it would be safer to explicitly define the domain in a `let` and update the code that inits `WordPressOrgXMLRPCApiError` to set it?

Reply:
My preference would be going with the safer option, which is avoiding breaking changes. You can implement `CustomNSError` to set a custom domain, which means we shouldn't need to change this assertion here.

Reply:
Partially addressed in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/782/commits/3f578954acf42bb50699d1885779f7d947e609fd. However, the tests still fail when build for SPM. I believe that has to do with the XML RPC returning `WordPressOrgXMLRPCApiError` or `WordPressOrgXMLRPCApiFault`. I'll have to look into the latter as well, I guess.

Reply:
I found the source of the issue.

If one removes the `@objc` annotation, then the `CustomNSError` conversion code is used in its entirety. Otherwise, with the error `enum` declared as `@objc`, it seems like only the part that computes `userInfo` is called.

I verified this with an ad hoc package. I might put it online later. I'm unsure how to move forward in the meantime, because we do need the `@objc` visibility for `WordPressOrgXMLRPCApiError`. I'll keep thinking about it...

Reply:
Move my WIP work to https://github.com/wordpress-mobile/WordPressKit-iOS/pull/790 and rebased it off this branch.